File: back_end/src/main/java/com/example/back_end/core/admin/category/service/CategoryService.java
package com.example.back_end.core.admin.category.service;

import com.example.back_end.core.admin.category.payload.request.CategoryRequest;
import com.example.back_end.core.admin.category.payload.response.CategoryNameResponse;
import com.example.back_end.core.admin.category.payload.response.CategoryResponse;
import com.example.back_end.core.common.PageResponse;

import java.util.List;

public interface CategoryService {

    void createCategory(CategoryRequest categoryRequest);

    void updateCategory(Long id, CategoryRequest request);

    PageResponse<?> getAll(String name, Boolean published, Integer pageNo, Integer pageSize);

    CategoryResponse getCategory(Long id);

    void deleteCategories(List<Long> ids);

    List<CategoryNameResponse> getCategoriesName();

}

File: front_end/src/admin/apis/CategoryService.ts
import { http } from '@/libs/http'
import {
    CategoryFilter,
    CategoryRequest,
    CategoryResponse,
    CategoryResponseBasic,
    CategoryResponseWithPage,
} from '@/model/Category'

class CategoryService {
    async getAll(filter: CategoryFilter) {
        const queryParams = new URLSearchParams({
            name: filter.name ?? '',
            pageNo: filter.pageNo?.toString() ?? '1',
            pageSize: filter.pageSize?.toString() ?? '6',
            published: filter.published?.toString() ?? '',
        })
        const url = `/admin/categories?${queryParams.toString()}`
        const result = await http.get<CategoryResponseWithPage>(url)
        return result.payload
    }

    async get(id: number) {
        const url = `/admin/categories/${id}`
        const result = await http.get<{
            data: CategoryResponse
        }>(url)
        return result.payload.data
    }

    async create(request: CategoryRequest) {
        const url = '/admin/categories'
        const result = await http.post<CategoryResponseBasic>(url, request)
        return { status: result.payload.status, message: result.payload.message }
    }

    async update(request: Partial<CategoryRequest>) {
        const url = `/admin/categories/${request.id}`
        const result = await http.put<CategoryResponseBasic>(url, request)
        return { status: result.payload.status, message: result.payload.message }
    }

    async delete(ids: number[]) {
        const url = '/admin/categories'
        const result = await http.delete<CategoryResponseBasic>(url, ids)
        return { status: result.payload.status, message: result.payload.message }
    }
}

export default new CategoryService()
